sample-resource-plugin/src/main/java/org/opensearch/sample/resource/actions/rest/create/CreateResourceRestAction.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sample.resource.actions.rest.create;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.sample.SampleResource;
+import org.opensearch.transport.client.node.NodeClient;
+
+import static org.opensearch.rest.RestRequest.Method.POST;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+import static org.opensearch.sample.utils.Constants.SAMPLE_RESOURCE_PLUGIN_API_PREFIX;
+
+/**
+ * Rest Action to create a Sample Resource. Registers Create and Update REST APIs.
+ */
+public class CreateResourceRestAction extends BaseRestHandler {
+
+    public CreateResourceRestAction() {}
+
+    @Override
+    public List<Route> routes() {
+        return List.of(
+            new Route(PUT, SAMPLE_RESOURCE_PLUGIN_API_PREFIX + "/create"),
+            new Route(POST, SAMPLE_RESOURCE_PLUGIN_API_PREFIX + "/update/{resource_id}")
+        );
+    }
+
+    @Override
+    public String getName() {
+        return "create_update_sample_resource";
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        Map<String, Object> source;
+        try (XContentParser parser = request.contentParser()) {
+            source = parser.map();
+        }
+
+        switch (request.method()) {
+            case PUT:
+                return createResource(source, client);
+            case POST:
+                return updateResource(source, request.param("resource_id"), client);
+            default:
+                throw new IllegalArgumentException("Illegal method: " + request.method());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
```

Comment:
Using `@SuppressWarnings("unchecked")` on untrusted input (the request body source) bears the risk of getting `ClassCastExceptions` deep inside the application code. While this is usually not security critical, it might locating the root cause of issues diificult - both for admins and for end users.

Reply:
I agree. However this is is a sample plugin and hence should be fine IMO.

Reply:
Well, the primary purpse of sample plugins is to give other developers an idea/blueprint on how to design their own code. Thus, there is a big chance that developers will follow any pattern displayed in that code base, including bad patterns.

And, experience shows that developers have the tendency to follow patterns they are finding in existing code bases, no matter how bad these actually might be.

IMHO, to cope for this, we should add in such cases at least a comment like "this is just done for simplicity, do not do that for real plugins".

Reply:
We do use this pattern at a decent number of places in security plugin. Here is one such [example](https://github.com/opensearch-project/security/blob/3c635c9f64ec58be206d9c43a865bd8cfa423fa1/src/main/java/org/opensearch/security/privileges/dlsfls/IndexToRuleMap.java#L57). I will add a comment here stating suppressing should be avoided in real-world, however there is no mechanism to stop plugin's from implementing it.

Reply:
There are different "risk classes" of using `@SuppressWarnings("unchecked")`. For the example you have linked, it is possible to argue within a minute that this is actually a safe cast and no late `ClassCastExceptions` can be caused by it.